third_party/amd/lib/TritonAMDGPUTransforms/BlockPingpong.cpp

```diff
@@ -81,6 +83,22 @@ void Pingponger::appendOp(Operation *op) {
   op->moveAfter(lastInsertedOp);
   lastInsertedOp = op;
 }
+void Pingponger::moveOpAndDependencies(Operation *op) {
```

Comment:
Can you please add comments explaining what this function does?
I suppose it fixes broken dependency order by pulling up ops which were originally located before the currently appending op.
This function would be very useful here and nice if we have it documented.